src/node/txpackagetracker.cpp

```diff
@@ -0,0 +1,547 @@
+// Copyright (c) 2022
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <node/txpackagetracker.h>
+
+#include <common/bloom.h>
+#include <logging.h>
+#include <policy/policy.h>
+#include <txorphanage.h>
+#include <txrequest.h>
+#include <util/hasher.h>
+
+namespace {
+    /** How long to wait before requesting orphan ancpkginfo/parents from an additional peer.
+     * Same as GETDATA_TX_INTERVAL. */
+    static constexpr auto ORPHAN_ANCESTOR_GETDATA_INTERVAL{60s};
+}
+class TxPackageTracker::Impl {
+    /** Manages unvalidated tx data (orphan transactions for which we are downloading ancestors). */
+    TxOrphanage& orphanage_ref;
+
+    mutable Mutex m_mutex;
```

Comment:
Perhaps:

```suggestion
    mutable Mutex m_package_tracker_mutex;
```

to help with grepping.